tests/cpp/test_multidevice_sharding.cpp

```diff
@@ -230,4 +230,55 @@ TEST_F(MultideviceShardingTest, Issue2758) {
       __FILE__);
 }
 
+TEST_F(MultideviceShardingTest, Broadcast) {
```

Comment:
Could you add some comments here? For example, what is this test meant to verify? How would this test fail (or not?) if the change of this PR were not done?

Reply:
Done.

Reply:
So, before this PR, if this test is executed with, say, 4 GPUs, the extent of `in->getLogicalDomain()[0]->extent()` and `out->getLogicalDomain()[0]->extent()` would be bound to 4? I'm not sure what would happen then, but I understand that isn't what it should be.

Reply:
> would be bound to 4?

... would be attempted to, and then hit something like "Tried to bind to a constant value: 1 as 4". Overall, we haven't tested a multidevice fusion with broadcasted inputs. Therefore, the bug and this PR.